### ISSUE TITLE
Error in r0 to dl1 step

### ISSUE THREAD

**Issue by morcuended** [cta-observatory/cta-lstchain#518]
When processing run 02469 subrun 85 from 2020_08_17 using the current master version (`v0.6.0.post68+git43951ff`) I ran into this error (which did not happen with v0.6.0)

```
Error writing col "intensity" of container "DL1ParametersContainer"
Traceback (most recent call last):
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/bin/lstchain_data_r0_to_dl1", line 8, in <module>
    sys.exit(main())
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/lstchain/scripts/lstchain_data_r0_to_dl1.py", line 163, in main
    tib_counter0=args.tib_counter0
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/lstchain/reco/r0_to_dl1.py", line 674, in r0_to_dl1
    containers = [dl1_container])
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/ctapipe/io/hdf5tableio.py", line 310, in write
    self._append_row(table_name, containers)
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/ctapipe/io/hdf5tableio.py", line 279, in _append_row
    value = self._apply_col_transform(table_name, colname, value)
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/ctapipe/io/tableio.py", line 118, in _apply_col_transform
    value = tr(value)
  File "/home/daniel.morcuende/.local/miniconda3/envs/osa/lib/python3.7/site-packages/ctapipe/io/hdf5tableio.py", line 465, in tr_convert_and_strip_unit
    return quantity.to_value(unit)
AttributeError: 'numpy.float64' object has no attribute 'to_value'

```

command issued:

```
lstchain_data_r0_to_dl1 -f /fefs/aswg/data/real/R0/20200817/LST-1.1.Run02469.0085.fits.fz \
     -p /fefs/aswg/workspace/daniel.morcuende/data/real/calibration/20200817/v05/drs4_pedestal.Run2460.0000.fits \
     --calib /fefs/aswg/workspace/daniel.morcuende/data/real/calibration/20200817/v05/calibration.Run2462.0000.hdf5 \
     -t /fefs/aswg/workspace/daniel.morcuende/data/real/calibration/20200817/v05/time_calibration.Run2462.0000.hdf5 \
     --pointing /fefs/onsite/monitoring/driveLST1/DrivePositioning/drive_log_20_08_17.txt \
     --ucts-t0-dragon 1597726889289278213 --dragon-counter0 4289277900 \
     --ucts-t0-tib 1597726889289278213 --tib-counter0 4289277500
```

**Comment by morcuended** [comment]
Same happens with run 02289 subrun 39

**Comment by moralejo** [comment]
Strange. Why did this not happen before?  "intensity" has no units, so if it is treated as an astropy quantity it will indeed make trouble - but we have not changed the ctapipe version (it is the latest release, 0.8.0, correct?), so why should it fail only now?

**Comment by morcuended** [comment]
> Strange. Why did this not happen before? "intensity" has no units, so if it is treated as an astropy quantity it will indeed make trouble - but we have not changed the ctapipe version (it is the latest release, 0.8.0, correct?), so why should it fail only now?

Yes, ctapipe version is 0.8.0. It is strange because it has happened only for these two subruns so far (haven't tested it with all the dates)

**Comment by moralejo** [comment]
Then may be https://github.com/cta-observatory/cta-lstchain/pull/510, some change of behaviour of astropy?

**Comment by moralejo** [comment]
> Then may be #510, some change of behaviour of astropy?

No, @morcuended tested it and it is not related.

**Comment by moralejo** [comment]
I think this is  connected to the fact that we now write out all events, regardless of whether the cleaning+parametrization step is successful or not. Possibly this is happening when some value (like intensity) has a weird value, like e.g. nan?

**Comment by moralejo** [comment]
The problem occurs when the **first event** in a run has 0 pixels after cleaning (an interleaved pedestal). In that case some default values are filled in DL1ParametersContainer. The **next** event then fails with the error you report. I think it is due to the fact that the data types of the container are not defined until the first time it is properly filled, when the first event with n_pixels>0 arrives.
Since this DL1 data format is deprecated, and we need to have a release now, I will just fix this by skipping the first event in a run if it happens to have 0 pixels after cleaning.

**Comment by moralejo** [comment]
Should be fixed in https://github.com/cta-observatory/cta-lstchain/pull/517

**Comment by MaxNoe** [comment]
> DL1ParametersContainer

This is exactly what I commented on here: https://github.com/cta-observatory/cta-lstchain/pull/497/files#diff-72698bec55fa154a478bc403994ed27bR164

How fast that "theoretical point of view" became an actual bug.

So please, don't add another workaround for the workaround. Fix the container defaults to have correct values and units.

**Comment by moralejo** [comment]
> > DL1ParametersContainer
> 
> This is exactly what I commented on here: https://github.com/cta-observatory/cta-lstchain/pull/497/files#diff-72698bec55fa154a478bc403994ed27bR164
> 
> How fast that "theoretical point of view" became an actual bug.

Well, the discussion there was about whether it mattered, for events in which **image was not parameterized**, to have 0's instead of nans as defaults. And I still think it does not matter. A different thing is whether another property of the variables (type, units) prevents events to be written out, which is what we found now.  Of course that matters. And yes, probably it would be fixed by a more careful setting of the container's defaults, but I think making the release has priority now. Making a proper solution will take more time because it may bring about other unexpected secondary effects.

**Comment by moralejo** [comment]
Addressed in #523 along the lines (hopefully) suggested by @MaxNoe, i.e. initializing the container with nan (instead of None's) for parameters that should always be there.

**Comment by rlopezcoto** [comment]
Solved in #526
